turbinia/client.py

```diff
@@ -0,0 +1,229 @@
+    """Prints the recent history for Turbinia Tasks.
+
+    Args:
+      instance (string): The Turbinia instance name (by default the same as the
+          PUBSUB_TOPIC in the config).
+      project (string): The name of the project.
+      region (string): The name of the zone to execute in.
+      days (int): The number of days we want history for.
+      task_id (string): The Id of the task.
+      request_id (string): The Id of the request we want tasks for.
+
+    Returns:
+      List of Task dict objects.
+    """
+    function = GoogleCloudFunction(project_id=project, region=region)
+    func_args = {'instance': instance, 'kind': 'TurbiniaTask'}
+
+    if days:
+      start_time = datetime.now() - timedelta(days=days)
+      # Format this like '1990-01-01T00:00:00z' so we can cast it directly to a
+      # javascript Date() object in the cloud function.
+      start_string = start_time.strftime('%Y-%m-%dT%H:%M:%S')
+      func_args.update({'start_time': start_string})
+    elif task_id:
+      func_args.update({'task_id': task_id})
+    elif request_id:
+      func_args.update({'request_id': request_id})
+
+    response = function.ExecuteFunction(function_name, func_args)
+    if not response.has_key('result'):
+      log.error('No results found')
+      print '\nNo results found.\n'
+      if response.get('error', '{}') != '{}':
+        msg = 'Error executing Cloud Function: [{0!s}].'.format(
+            response.get('error'))
+        print '{0:s}\n'.format(msg)
+        log.error(msg)
+      log.debug('GCF response: {0!s}'.format(response))
+      raise TurbiniaException(
+          'Cloud Function {0:s} returned no results.'.format(function_name))
+
+    try:
+      results = json.loads(response['result'])
+    except (TypeError, ValueError) as e:
+      raise TurbiniaException(
+          'Could not deserialize result from GCF: [{0!s}]'.format(e))
+
+    return results[0]
+
+
+  def print_task_status(self, instance, project, region, days=0, task_id=None,
+                        request_id=None, all_fields=False):
+    """Prints the recent history for Turbinia Tasks.
+
+    Args:
+      instance (string): The Turbinia instance name (by default the same as the
+          PUBSUB_TOPIC in the config).
+      project (string): The name of the project.
+      region (string): The name of the zone to execute in.
+      days (int): The number of days we want history for.
+      task_id (string): The Id of the task.
+      request_id (string): The Id of the request we want tasks for.
+      all_fields (bool): Print all fields for the task, including task, request
+          ids and saved file paths.
+    """
+    task_results = self.get_task_data(instance, project, region, days, task_id,
+                                      request_id)
+    num_results = len(task_results)
+    if not num_results:
+      print '\nNo Tasks found.'
+      return
+
+    print '\nRetrieved {0:d} Task results:'.format(num_results)
```

Comment:
The reason I originally wanted to have prints here instead of logging was so that I could control the output format a little more than with just logging.  I changed this to be format_task_status() instead, and do the print in turbiniactl.